File: tests/src/main/java/com/github/klee0kai/test/lifecycle/structure/Ram.java
package com.github.klee0kai.test.lifecycle.structure;

import com.github.klee0kai.test.lifecycle.di.qualifiers.RamSize;

import java.util.UUID;

public class Ram {

    public final UUID uuid = UUID.randomUUID();

    public final String size;

    public Ram() {
        size = null;
    }

    public Ram(RamSize ramSize) {
        this.size = ramSize.size;
    }
}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/lifecycle/structure/Ram.kt
package com.github.klee0kai.test.lifecycle.structure

import com.github.klee0kai.test.lifecycle.di.qualifier.RamSize
import java.util.*

class Ram {
    val uuid = UUID.randomUUID()
    val size: String?

    constructor() {
        size = null
    }

    constructor(ramSize: RamSize) {
        size = ramSize.size
    }
}